staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -779,7 +779,7 @@ func growSlice(v reflect.Value, maxCapacity int, sizes ...int) {
 		return
 	}
 	if v.Len() > 0 {
-		extra := reflect.MakeSlice(v.Type(), 0, max)
+		extra := reflect.MakeSlice(v.Type(), v.Len(), max)
```

Comment:
This is kind of a terrifying function and it needs a bunch of unit tests. ... it looks to have been broken quite badly in ways that should cause drastic problems everywhere it's used, so either it's barely ever called, or we're reading it wrong, but either way I need to see a unit test.

Reply:
@logicalhan do you know anything about this function?

Reply:
> so either it's barely ever called

I agree with this opinion. I have tested several versions and only the >= 1.24 version has errors.

The current exposure of this problem depends on https://github.com/kubernetes/kubernetes/issues/108569's code modifications and  https://github.com/kubernetes/kubernetes/issues/110625's feedback. (Thanks @mborsz and @yokomotod!) 

@yokomotod's test case is typical, in fact, if only some objects with the same lables were created, this problem would not be reproduced, otherwise we would not have found a problem here.

Maybe other scenarios also have the problem, but we did not find the feedback

Reply:
Can you open an issue to capture these two points? (I won't block this PR on them since the bug looks disastrous)
* I am skeptical that this function is actually an improvement over just letting the go runtime pick the size. I'd like to see benchmarks with it swapped out for just calling append.
* If we do keep it, this desperately needs to be rewritten to use the new type parameters instead of reflection.